Sources/DockProgress/DockProgress.swift

```diff
@@ -250,3 +262,20 @@ public enum DockProgress {
 		}
 	}
 }
+
+private func lerp(_ start: Double, _ end: Double, _ t: Double) -> Double {
+    return start + (end - start) * t
+}
+
+private func easeIn(_ t: Double) -> Double {
+    // Smoothstep, i.e. approximately iOS's default easing function (see https://stackoverflow.com/a/25728953)
+    return 3 * pow(t, 2) - 2 * pow(t, 3);
+}
+
+private func easeOut(_ t: Double) -> Double {
+    return 1 - easeIn(1 - t)
+}
+
+private func easeInOut(_ t: Double) -> Double {
+    return lerp(easeIn(t), easeOut(t), t)
+}
```

Comment:
I would be nicer to wrap these into a namespace and move them to utilities.swift. Not sure if it's more efficient, but maybe we could also use SIMD for this:

```swift
import simd

enum Easing {
    static func lerp(_ start: Double, _ end: Double, _ t: Double) -> Double {
        return Double(simd_mix(Float(start), Float(end), Float(t)))
    }

    static private func easeIn(_ t: Double) -> Double {
        return Double(simd_smoothstep(0.0, 1.0, Float(t)))
    }

    static private func easeOut(_ t: Double) -> Double {
        return 1 - easeIn(1 - t)
    }

    static func easeInOut(_ t: Double, _ duration: Double) -> Double {
        let normalizedTime = t / duration
        return lerp(easeIn(normalizedTime), easeOut(normalizedTime), normalizedTime)
    }
}
```